features/eyetracker_features.py

```diff
@@ -325,14 +326,37 @@ def __init__(self, *args, **kwargs): #, start_pos, calibration):
 
             # calculate coefficients to calibrate eye data
             events = bmi3d_data['events']
-            self.eye_coeff,_,_,_ = aopy.preproc.calc_eye_calibration\
-                (bmi3d_data['cursor_interp'],bmi3d_metadata['cursor_interp_samplerate'],\
-                eye_interp[:,:4], bmi3d_metadata['cursor_interp_samplerate'],events['timestamp'], events['code'],return_datapoints=True)
 
+            if not self.eye_target_calibration:
+                self.eye_coeff,_,_,_ = aopy.preproc.calc_eye_calibration\
+                    (bmi3d_data['cursor_interp'],bmi3d_metadata['cursor_interp_samplerate'],\
+                    eye_interp[:,:4], bmi3d_metadata['cursor_interp_samplerate'],events['timestamp'], events['code'],return_datapoints=True)
+
+            else:
+                def get_target_locations(data, target_indices):
+
+                    try:
+                        trials = data['trials']
+                    except:
+                        trials = data['bmi3d_trials']
+                    locations = np.nan*np.zeros((len(target_indices), 3))
+                    for i in range(len(target_indices)):
+                        trial_idx = np.where(trials['index'] == target_indices[i])[0]
+                        if len(trial_idx) > 0:
+                            locations[i,:] = trials['target'][trial_idx[0]][[0,2,1]] # use x,y,z format
+                        else:
+                            raise ValueError(f"Target index {target_indices[i]} not found")
+                    return np.round(locations,4)
+
+                target_pos = get_target_locations(bmi3d_data, [1,2,3,4,5,6,7,8])
```

Comment:
how do you know it's going to be targets 1-8? should you check the event codes?

Reply:
This calibration works only for 8 target center out data. This needs to be modified in the future.